src/com/company/Main.java

```diff
@@ -0,0 +1,76 @@
+package com.company;
+
+import com.company.faculty.Faculty;
+import com.company.faculty.FacultyFactory;
+import com.company.group.Group;
+import com.company.student.Student;
+import com.company.subject.English;
+import com.company.subject.Mathematics;
+import com.company.subject.TheoreticalComputerScience;
+import com.company.university.University;
+
+import java.util.Objects;
+
+public class Main {
+
+    public static void main(String[] args) {
+        try {
+            University university = University.getInstance();
+            System.out.println();
+            System.out.println("The university has two faculties: ");
+            Faculty mathematics = null;
+            try {
+                mathematics = FacultyFactory.getFaculty("Mathematics");
+                Group group1 = new Group("Group1");
+
+                mathematics.getGroups().add(group1);
+
+                try {
+                    Student student1 = new Student.SubjectBuilder()
+                            .addEnglish(8)
+                            .addHistory(4)
+                            .addMathematics(6)
+                            .addPhilosophy(7)
+                            .addTheoreticalComputerScience(5)
+                            .build();
+                    student1.setFirstName("Ashot ");
+                    student1.setLastName("Amaryan");
+                    group1.getStudents().add(student1);
+                    System.out.print("the average of scores is:");
+                    System.out.println(student1.averageOfStudentScores());
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+
+                try {
+                    System.out.println();
+                    Student student2 = new Student.SubjectBuilder()
+                            .addFrench(5)
```

Comment:
Diana jan, I have some questions here. :)) At first there were a requirement to have same subjects for groups, so I created Builder in the class Group, then two days before the requirement was changed, because it is the student  who must choose subjects, so I have rewritten builder for student class and changed all my code, that was the reason to send you my project later. I can do the same thing for the Faculty too, but I want to understand definitely the logic of this problem. Because in the requirement there were nothing about it. But we must have an exception in case if the student have no subject, as I understand, the student can choose subject even if the subject  doesn't exist in his faculty. Subjects, in general, in Armenia :) are the same for professions, so called groups, but the faculty can have different professions, different groups, but I think that the model when student choose subjects is more flexible. It's the European university system. I understand that it's strange to build a student according to the subjects he choose, but here it's necessary because student must have at least one subject. How do you think about it? ^_^

Reply:
@Lusin-mus  jan I am sorry you are the second student who already told me that requirement was to not specify subjects for groups, then for students, and I was not aware of that requirement. In that case, you can leave it as it is "European university system"), but my point here also is that Student can't be created already with some marks, you can use same builder patterns to just assign initial list of subjects to student, then provide methods to add new subjects and  add marks for his particular subject

Reply:
Diana jan, now I understand what you mean :) Excuse-me I couldn't work yesterday, but today I will try to add new methods :) In that case, I will change the exceptions too, because the logic of the project changes. Thanks you much ^_^

Reply:
My dear Diana, finely I finished (I hope so) the project, I had some difficulties, but I think I could do what you suggested :) Thank you a lot ^_^